internal/v4/api_test.go

```diff
@@ -1586,89 +1577,446 @@ func (s *APISuite) TestResolveURL(c *gc.C) {
 	}
 }
 
+var serveExpandIdCharms = []struct {
+	id    *router.ResolvedURL
+	charm charm.Charm
+}{{
+	id: newResolvedURL("cs:~charmers/precise/wordpress-0", 0),
+	// Note: this charms sets the ACL for all ~chamers/wordpress charms.
+	charm: testCharm{
```

Comment:
I implemented the thing at the lowest level that if it's nil it uses a storetesting.NewCharm(nil) to make this all tickety-boo.